### PR TITLE
Pool: adds a test for the bug in question, no weird behaviour allthough I used pretty similar values with what's already present in the live environment

### PR DESCRIPTION
**Pull request by gangov**

### CODE REVIEW

File: contracts/pool/src/tests/swap.rs [hunk 1]
```diff
@@ -639,3 +639,76 @@ fn swap_simulation_one_third_pool() {
...
+    env.mock_all_auths();
+    env.budget().reset_unlimited();
+
+    let mut token1 = deploy_token_contract(&env, &Address::random(&env));
+    let mut token2 = deploy_token_contract(&env, &Address::random(&env));
+    if token2.address < token1.address {
+        std::mem::swap(&mut token1, &mut token2);
+    }
+
+    let swap_fees = 1_000i64; // 10% bps
```
Comment (ueco-jb) [inline contracts/pool/src/tests/swap.rs:655]: Can you try smaller amounts, 1% and 0.3%?

Comment (gangov) [reply contracts/pool/src/tests/swap.rs:655]: got the expected results

**Review by ueco-jb — COMMENTED** [review]
I like those tests. Instead of creating a new, specifically named test cases let's close this withing one `test_swap_fee_variants` or something; use `testcase` crate and add that fee as parameter to the tests in couple variants, including those 1%, 0.3%.
https://docs.rs/test-case/latest/test_case/

File: contracts/pool/src/tests/swap.rs [hunk 2]
```diff
@@ -756,83 +685,17 @@ fn bug_issue_169_1_percent() {
...
-
-    let mut token1 = deploy_token_contract(&env, &Address::random(&env));
-    let mut token2 = deploy_token_contract(&env, &Address::random(&env));
-    if token2.address < token1.address {
-        std::mem::swap(&mut token1, &mut token2);
+    if swap_fees < 100 {
+        fees = Decimal::bps(fee_percentage) * output_amount;
+    } else {
+        fees = Decimal::percent(fee_percentage) * output_amount;
     }
```
Comment (ueco-jb) [inline contracts/pool/src/tests/swap.rs]: Keep fees in BPS, adjust parameters from test decorator. 1% is 100 etc.

```suggestion
    fees = Decimal::bps(fee_percentage) * output_amount;
```

File: contracts/pool/src/tests/swap.rs [hunk 3]
```diff
@@ -640,81 +640,10 @@ fn swap_simulation_one_third_pool() {
...
-            spread_amount: Decimal::from_ratio(offer_amount, initial_liquidity) * output_amount,
-            commission_amount: fees,
-        }
-    );
-}
-
-#[test]
-fn bug_issue_169_1_percent() {
+#[test_case(1_000i64, 10, 99102002 ; "when fee is 10%")]
+#[test_case(100, 1, 9910200 ; "when fee is 1%")]
```
Comment (ueco-jb) [inline contracts/pool/src/tests/swap.rs:644]:
```suggestion
#[test_case(1_000i64, 1000, 99102002 ; "when fee is 10%")]
#[test_case(100, 100, 9910200 ; "when fee is 1%")]
```

File: contracts/pool/src/tests/swap.rs [hunk 4]
```diff
@@ -843,12 +706,12 @@ fn bug_issue_169_below_1_percent() {
         }
     );
 
-    // 991020025 is the request, so 0.3% of that should be what's on the left hand side
-    assert_eq!(2973060, result.commission_amount);
+    // 991020025 is the request, so 10% of that should be what's on the left hand side
```
Comment (ueco-jb) [inline contracts/pool/src/tests/swap.rs:707]: Those comments alre aren't accurate in parametrized test.

File: contracts/pool/src/tests/swap.rs [hunk 5]
```diff
@@ -843,12 +706,12 @@ fn bug_issue_169_below_1_percent() {
...
+    assert_eq!(commission_fee, result.commission_amount);
 
     let result = pool.simulate_reverse_swap(&token1.address, &(output_amount - fees));
     let output_amount = 991020025i128;
-    let fees = Decimal::bps(30) * output_amount;
+    // let fees = Decimal::percent(fee_percentage) * output_amount;
```
Comment (ueco-jb) [inline contracts/pool/src/tests/swap.rs:712]: Some leftover?

```suggestion
```

**Review by ueco-jb — COMMENTED** [review]
Good test, just some cleaning required.

File: contracts/pool/src/tests/swap.rs [hunk 6]
```diff
@@ -725,7 +654,7 @@ fn bug_issue_169_1_percent() {
         std::mem::swap(&mut token1, &mut token2);
     }
 
-    let swap_fees = 1_00i64; // 1% bps
+    let swap_fees = swap_fees; // 10% bps
```
[the remark this answers is not in the document]
Comment (gangov) [reply contracts/pool/src/tests/swap.rs]: the linter also complained about this one, it's not in the code anymore

File: contracts/pool/src/tests/swap.rs (hunk 3 above)
Comment (gangov) [reply contracts/pool/src/tests/swap.rs]: only using BPS

**Review by ueco-jb — APPROVED** [review]
Good test :+1:

**Pull request merged**